annotated_test.go

```diff
@@ -699,6 +705,78 @@ func TestAnnotatedAs(t *testing.T) {
 			},
 			startApp: true,
 		},
+		{
+			desc: "self w other As annotations",
+			provide: fx.Provide(
+				fx.Annotate(
+					func() *asStringer {
+						return &asStringer{name: "stringer"}
+					},
+					fx.As(fx.Self()),
+					fx.As(new(fmt.Stringer)),
+				),
+			),
+			invoke: func(s fmt.Stringer, as *asStringer) {
+				assert.Equal(t, "stringer", s.String())
+				assert.Equal(t, "stringer", as.String())
+			},
+		},
+		{
+			desc: "self as one As target",
+			provide: fx.Provide(
+				fx.Annotate(
+					func() (*asStringer, *bytes.Buffer) {
+						s := &asStringer{name: "stringer"}
+						b := &bytes.Buffer{}
+						return s, b
+					},
+					fx.As(fx.Self(), new(io.Writer)),
```

Comment:
I'm sure this still works, and there's already a test case that covers this, but could we add another case with exactly this and the following annotation added:
```
fx.As(fmt.Stringer)
```

This should provide four types in total:
* *asStringer
* io.Writer
* fmt.Stringer
* *bytes.Buffer